warehouse/client/controlplane/client.go

```diff
@@ -0,0 +1,99 @@
+package controlplane
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/rudderlabs/rudder-server/utils/httputil"
+)
+
+type PublicPrivateKeyPair struct {
+	PublicKey  string
+	PrivateKey string
+}
+
+type BasicAuth struct {
+	Username string
+	Password string
+}
+
+type InternalControlPlane interface {
+	GetDestinationSSHKeys(ctx context.Context, id string) (*PublicPrivateKeyPair, error)
+}
+
+type internalClientWithCache struct {
+	client InternalControlPlane
+	cache  map[string]*PublicPrivateKeyPair
+}
+
+func NewInternalClient(baseURI string, auth BasicAuth) InternalControlPlane {
+	return &internalClient{
+		baseURI: baseURI,
+		auth:    auth,
+	}
+}
+
+type internalClient struct {
+	baseURI string
+	auth    BasicAuth
+}
+
+func (api *internalClient) GetDestinationSSHKeys(ctx context.Context, id string) (*PublicPrivateKeyPair, error) {
+	url := fmt.Sprintf("%s/dataplane/admin/destinations/%s/sshKeys", api.baseURI, id)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, http.NoBody)
+	if err != nil {
+		return nil, fmt.Errorf("creating new request with ctx: %w", err)
+	}
+
+	req.SetBasicAuth(api.auth.Username, api.auth.Password)
+	req.Header.Set("Content-Type", "application/json")
+
+	client := http.Client{}
```

Comment:
Avoid creating a new client in every request, cause it is going to establish a new tcp connection every time. 

Instead, reuse the client to enable connection pooling. We can simply do:

```go
type internalClient struct {
	baseURI string
	auth    BasicAuth
	client *http.Client
}
```

During the construction, you should also set the timeout.

```suggestion
```

Reply:
That's a good point about reusing the tcp connection. I'll make the change now.